dev/core#2420 Add text to report template saying no results found if that is the case

eileenmcnaughton opened the pull request:
Overview
----------------------------------------
Adds explanatory text in place of results when a report returns no results

Before
----------------------------------------
![before](https://user-images.githubusercontent.com/336308/109080515-9929fc80-7765-11eb-82b4-26e463e60d0d.png)

After
----------------------------------------
![image](https://user-images.githubusercontent.com/336308/109086439-7224f800-7770-11eb-98f5-b26dd48f5fdf.png)


Technical Details
----------------------------------------
This is very much the minimum required to reduce confusion & leave all worms in their cans

Note this is done to create minimum white space change. I actuallY
think the criteria around charts should be in the tpl that
makes the decision to include Table.tpl rather than in Table.tpl
and this would become {if !$rows}{/else}{ts}No results found{/ts}{/if}

I can do that as a follow up if agreed

Comments
----------------------------------------

eileenmcnaughton commented:
@MegaphoneJon @yashodha feels like the sort of thing you might have ideas on

eileenmcnaughton commented:
Just noting a +1 by @demeritcowboy on https://lab.civicrm.org/dev/core/-/issues/2293#note_52642

demeritcowboy commented:
Two things I notice:
* There should be brackets around the `||` group, otherwise on some reports, e.g. Case Details or Event Participants, you get both "no results found" and the actual results.
* On a brand new civireport from template, before you've clicked refresh, it says "no results found" whereas it's more like you just haven't run the report yet. While that distinction is obvious to experienced users it might not be obvious to newcomers. But I'm not sure how to avoid that if it's in here. Hmm.

eileenmcnaughton commented:
@demeritcowboy "empty result set" or "no results retrieved"?

demeritcowboy commented:
It outputs "no results found" on the screen.

eileenmcnaughton commented:
@demeritcowboy yep - I mean - what would be better

MegaphoneJon commented:
I don't have many thoughts other than that this is a good idea.

colemanw commented:
In general I try to avoid adding new strings for translation when I can avoid it. A grep of "No results found" gives, appropriately, no results found.

Our standard reusable string for searches, reports & dashlets is "None found."

![image](https://user-images.githubusercontent.com/2874912/109162995-d59a3000-7746-11eb-9196-6ece651095c9.png)

eileenmcnaughton commented:
I've updated this to 'None Found' based on ^^

I've also addressed the issue @demeritcowboy pointed out about the parentheses by moving that if to the functions that determine whether to include table.tpl. IMHO you should call table.tpl if you want the table block and the decision about including it lives in the code that includes it

demeritcowboy commented:
**Issue**: My computer is being a super-nerd and treating `None found` different than `None found.` (with a period)

Otherwise looks ok to me and I don't think there's much that can be done easily when it's fresh from template. The underlying problem is that `$rows` is serving double-duty, so is kind of a separate issue.

demeritcowboy commented on templates/CRM/Logging/ReportDetail.tpl:
```diff
@@ -22,7 +22,7 @@
     {if $layout eq 'overlay'}
       {include file="CRM/Report/Form/Layout/Overlay.tpl"}
-    {else}
+    {elseif (!$chartEnabled || !$chartSupported)}
```
```suggestion
    {elseif (((((!$chartEnabled || !$chartSupported)))))}
```

seamuslee001 replied:
have you got enough brackets there @demeritcowboy ?

eileenmcnaughton replied:
Hmm I decided to go the another way & radically go bracketless

demeritcowboy replied:
Good choice.

eileenmcnaughton commented:
I added the .

demeritcowboy commented:
All good now.

eileenmcnaughton commented:
@colemanw can you merge based on @demeritcowboy's review?
